app/src/main/java/it/kapfer/digitalesamt/liberator/ModuleMain.kt

```diff
@@ -40,12 +40,10 @@ class ModuleMain : IXposedHookZygoteInit, IXposedHookLoadPackage {
     private fun handleBmf2Go(lpparam: XC_LoadPackage.LoadPackageParam) {
         XposedBridge.log("Hooking RootBeer")
         // Hook RootBeer's isRooted() method
-        XposedHelpers.findAndHookMethod(ROOTBEER_CLASS, lpparam.classLoader, "n", XC_MethodReplacement.returnConstant(false))
```

Comment:
I think, we should stay backwards compatible. With this change, older versions of the app don't work anymore, because installing the hook crashes it.

There seems to be a way to get the version of the app the module is loaded in: https://forum.xda-developers.com/t/get-app-version-before-hooking.2796761/post-85688177

So you could add a function that gets the version and determines by the returned version code which methods to hook:
```kotlin
    private fun getPackageVersion(lpparam: XC_LoadPackage.LoadPackageParam): Int {
        val apkPath = File(lpparam.appInfo.sourceDir)
        val packageParser = XposedHelpers.findClass("android.content.pm.PackageParser", lpparam.classLoader)
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            val packageLite = XposedHelpers.callStaticMethod(packageParser, "parsePackageLite", apkPath, 0)
            XposedHelpers.getIntField(packageLite, "versionCode")
        } else {
            val packageObject = XposedHelpers.callMethod(packageParser.newInstance(), "parsePackage", apkPath, 0)
            XposedHelpers.getIntField(packageObject, "mVersionCode")
        }
    }
```

The oldest version of the APK I could get a hold on was 3.0.0, which already has the unobfuscated method names you mention. This APK has version code **261**, so we should probably just start with that and assume the obfuscation was changed with the major release. But if you have more details on which version the change was introduced, you can use that.